### DISCUSSION TITLE
goto_pos from another buffer doesn't work?

### DISCUSSION THREAD

**Discussion by hckiang** [orbitalquark/textadept#129] [Q&A] [answered]
If you "focus" `_BUFFERS[1]` and run `buffer.goto_pos(_G._BUFFERS[3], 10)` and use Ctrl-Tab to switch to the buffer number 3, then you will see that the cursor in BUFFER number 3 has not been moved. I am wondering whether this is by design or a bug? Is there a way to change the cursor position of `_BUFFERS[3]` without "focusing" the target buffer?

**Comment by orbitalquark** [comment, marked as answer]
This is by design. Scintilla (Textadept's editing component) has a single view for all buffers and does not store per-buffer positional data. Textadept saves and restores some per-buffer data[1] when switching between views, so you can try setting those fields from your script. Calling buffer methods on the non-global buffer does not count as "switching views", so the save/restore event handlers are not invoked.

[1]: https://github.com/orbitalquark/textadept/blob/80816ecdba807af53f556dc3523b7726625bd5c4/core/ui.lua#L347

**Reply by hckiang** [reply]
Thanks! Now that makes sense!
